runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -275,13 +278,29 @@ else if (client.getWidget(WidgetInfo.QUEST_COMPLETED_NAME_TEXT) != null)
 			String text = client.getWidget(WidgetInfo.QUEST_COMPLETED_NAME_TEXT).getText();
 			fileName = "Quest(" + text.substring(19, text.length() - 1) + ")";
 		}
+		else if (hasBeenKilled)
+		{
+			hasBeenKilled = false;
+			takeScreenshot(client.getLocalPlayer().getName() + " Died on " + LocalDate.now());
+		}
 
 		if (fileName != null)
 		{
 			takeScreenshot(fileName);
 		}
 	}
 
+
+	@Subscribe
+	public void onLocalPlayerDeath(LocalPlayerDeath death)
+	{
+		if (config.screenshotPlayerDeads())
+		{
+			shouldTakeScreenshot = true;
```

Comment:
Why not just `takeScreenshot` here?

Reply:
It will do it double.

Reply:
Is LocalPlayerDeath fired twice on death? That doesn't sound right.

Reply:
Seems i made a mistake earlier, cant reproduce it so i changed it.